File: test_project/SampleViewController.m
#import "SampleViewController.h"

@implementation SampleViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // TODO: Add custom initialization
    [self setupCustomViews];
    
    /* TODO: Configure data source */
    [self configureDataSource];
}

- (void)setupCustomViews {
    // FIXME: Add proper constraints
    [self setupConstraints];
}

- (void)configureDataSource {
    // TODO: Implement data source methods
    self.dataSource = [[CustomDataSource alloc] init];
}

/* DONE: Add basic view setup */
- (void)setupBasicView {
    self.view.backgroundColor = [UIColor whiteColor];
}

@end

File: test_project/SampleViewController.swift
import Foundation

class SampleViewController {
    // TODO: Implement viewDidLoad method
    func viewDidLoad() {
        // TODO: Add navigation setup
        setupNavigation()
        
        // TODO: Configure table view
        configureTableView()
    }
    
    private func setupNavigation() {
        // FIXME: Add proper navigation title
        navigationItem.title = "Sample"
    }
    
    private func configureTableView() {
        // TODO: Set up table view delegate and data source
        tableView.delegate = self
        tableView.dataSource = self
    }
    
    // DONE: Add basic UI setup
    private func setupUI() {
        view.backgroundColor = .white
    }
}
